sql/core/src/main/scala/org/apache/spark/sql/DataFrameWriter.scala

```diff
@@ -447,7 +447,16 @@ final class DataFrameWriter[T] private[sql](ds: Dataset[T]) {
       // Create the table if the table didn't exist.
       if (!tableExists) {
         val schema = JdbcUtils.schemaString(df, url)
-        val sql = s"CREATE TABLE $table ($schema)"
+        // To allow certain options to append when create a new table, which can be
+        // table_options or partition_options.
+        // E.g., "CREATE TABLE t (name string) ENGINE=InnoDB DEFAULT CHARSET=utf8"
+        val createtblOptions = {
+          extraOptions.get("jdbc.create.table.options") match {
+            case Some(value) => " " + value
+            case None => ""
+          }
+        }
+        val sql = s"CREATE TABLE $table ($schema)" + createtblOptions
```

Comment:
Totally got your idea. Thanks a lot for the prompt response. I will update the patch.